patients/service/service.go

```diff
@@ -110,7 +113,17 @@ func (s *service) Update(ctx context.Context, update patients.PatientUpdate) (*p
 	}
 
 	s.logger.Infow("updating patient", "userId", existing.UserId, "clinicId", update.ClinicId)
-	return s.patientsRepo.Update(ctx, update)
+	patient, err := s.patientsRepo.Update(ctx, update)
+	if err != nil {
+		return nil, err
+	}
+
+	// Updates to the demo patient user should not affect the patient count
+	if update.UserId != s.config.ClinicDemoPatientUserId {
```

Comment:
RefreshPatientCounts allows for more than a single demo patient, but this code doesn't. Is there a reason for that asymmetry? I would expect this code to check for demo users in a similar fashion to RefreshPatientCounts.